e2e/testcases/kcc_test.go

```diff
@@ -129,3 +135,93 @@ func TestKCCResourcesOnCSR(t *testing.T) {
 		nt.T.Fatal(err)
 	}
 }
+
+func TestResourceGroupKCC(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.ACMController, ntopts.KCCTest)
+
+	ns := v1.Namespace{}
+	ns.Name = "resourcegroup-e2e"
+	nt.T.Cleanup(func() {
+		// all test resources are created in this namespace
+		if err := nt.KubeClient.Delete(ns.DeepCopy()); err != nil {
+			nt.T.Error(err)
+		}
+	})
+	if err := nt.KubeClient.Create(ns.DeepCopy()); err != nil {
```

Comment:
why deepcopy?

Reply:
I found it an easy abstraction to work with when writing the tests. Removed.

Reply:
Still got deepcopy used in a dozen other places in this file. It's probably not a big deal in terms of test time, but it's not very efficient. I'd rather get in the habit of avoiding deepcopy wherever possible.

Reply:
IMO this is such a low cost thing, particularly for a test case, that it's not worth the mental bandwidth optimizing for. But I removed the DeepCopy calls.

Reply:
Agree that it's not really worth optimizing for the speed alone. The other reason I should have mentioned was that it confused me when I read it and took me a while to understand. It's effectively a workaround to get around a gotcha that you have to know about in order to understand why it was done this way. So in this case not using deepcopy also helps improve readability. 

So thanks for cleaning it up.